Objects/complexobject.c

```diff
@@ -523,20 +519,13 @@ complex_pow(PyObject *v, PyObject *w, PyObject *z)
         return NULL;
     }
     errno = 0;
-    // Check if w is an integer value that fits inside a C long, so we can
-    // use a faster algorithm. TO_COMPLEX(w, b), above, already handled the
-    // conversion from larger longs, as well as other types.
-    if (PyLong_Check(w)) {
-        int overflow = 0;
-        long int_exponent = PyLong_AsLongAndOverflow(w, &overflow);
-        if (int_exponent == -1 && PyErr_Occurred())
-            return NULL;
-        if (overflow == 0)
-            p = c_powi(a, int_exponent);
-        else
-            p = _Py_c_pow(a, b);
-    } else {
-        p = _Py_c_pow(a, b);
+    exponent = b;
+    if (exponent.imag == 0.0 && exponent.real == rint(exponent.real)
```

Comment:
`rint` is standard in C99, so it should be safe, but we don't currently use it elsewhere in the codebase. We could equally well use `floor` or `ceil` - the behaviour would be the same, but the generated code may be less efficient, since on some processors `floor` and `ceil` require a rounding mode switch.

Reply:
Out of an overabundance of caution, I'm going to switch this to `floor`.